mono/metadata/gc.c

```diff
@@ -726,10 +726,18 @@ finalize_domain_objects (DomainFinalizationReq *req)
 #elif defined(HAVE_SGEN_GC)
 	while (!suspend_finalizers && (count = mono_gc_finalizers_for_domain (domain, to_finalize, NUM_FOBJECTS))) {
```

Comment:
@kumpera I agree. Do you want it to be part of this PR though ? I will revisit the entire finalization code anyway in the not too distant future.